nodestream/extractors/stores/aws/s3_extractor.py

```diff
@@ -14,30 +14,45 @@ def from_file_data(
         bucket: str,
         prefix: Optional[str] = None,
         object_format: Optional[str] = None,
+        archive_dir: str = None,
         **aws_client_args,
     ):
         return cls(
             bucket=bucket,
             object_format=object_format,
             prefix=prefix,
+            archive_dir=archive_dir,
             s3_client=AwsClientFactory(**aws_client_args).make_client("s3"),
         )
 
     def __init__(
         self,
         bucket: str,
+        archive_dir: str,
         s3_client,
         object_format: Optional[str] = None,
         prefix: Optional[str] = None,
     ) -> None:
         self.object_format = object_format
         self.prefix = prefix or ""
         self.bucket = bucket
+        self.archive_dir = archive_dir
         self.s3_client = s3_client
 
     def get_object_as_io(self, key: str) -> StringIO:
         return self.s3_client.get_object(Bucket=self.bucket, Key=key)["Body"]
 
+    def archive_s3_object(self, key: str):
+        if self.archive_dir:
+            self.logger.info("Archiving S3 Object", extra=dict(key=key))
```

Comment:
I don't think currently `S3Extractor` has a logger in here currently. I think its appropriate to add this one.

Reply:
Done